docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -185,25 +186,16 @@ And the following defaulting:
 
 - `KubeadmControlPlane.Spec.Replicas: 1`
 
-#### Modifications required to existing API Types
-
-- Add `Cluster.Spec.ControlPlaneRef` defined as:
+##### Changes for supporting different upgrade strategies
 
 ```go
-    // ControlPlaneRef is an optional reference to a provider-specific resource that holds
-    // the details for provisioning the Control Plane for a Cluster.
-    // +optional
-    ControlPlaneRef *corev1.ObjectReference `json:"controlPlaneRef,omitempty"`
-```
-
-- Add `Cluster.Status.ControlPlaneReady` defined as:
+type StrategyType string
```

Comment:
What about to use `UpgradeStrategyType` here, and rename `UpgradeStrategyType`  down below to `UpgradeStrategy`

Also, wondering if this should be generalized from `Upgrade` to `Rollout`, given that the same strategy applies no matter if the intent is to upgrade (change the Kubernetes version) or to apply any other change to the spec.
It will be great to have some more feedback here...

Reply:
I was thinking the same thing. +1 of using `Rollout` instead of `Upgrade`

Reply:
@fabriziopandini I updated the naming to use `Rollout` instead of `Upgrade`. IMO naming is now more generic and more in line with the whole proposal. 

One last thing came to my mind was should we use `ScaleDownStrategyType` and `ScaleUpStrategyType` instead of using `ScaleInStrategyType` and `ScaleOutStrategyType`?

Reply:
I have no strong opinion here, I'm ok with both options (just call the const ...RolloutType)